types/networks/src/lib.rs

```diff
@@ -302,3 +301,136 @@ impl Default for ChainConfig {
         }
     }
 }
+
+pub fn default_network_version() -> NetworkVersion {
+    NetworkVersion::V1
+}
+
+pub mod de_network_version {
+    use fil_types::NetworkVersion;
+    use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<NetworkVersion, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let version: &str = Deserialize::deserialize(deserializer)?;
+        let version = version.to_lowercase();
+
+        match version.as_str() {
+            "v0" => Ok(NetworkVersion::V0),
+            "v1" => Ok(NetworkVersion::V1),
+            "v2" => Ok(NetworkVersion::V2),
+            "v3" => Ok(NetworkVersion::V3),
+            "v4" => Ok(NetworkVersion::V4),
+            "v5" => Ok(NetworkVersion::V5),
+            "v6" => Ok(NetworkVersion::V6),
+            "v7" => Ok(NetworkVersion::V7),
+            "v8" => Ok(NetworkVersion::V8),
+            "v9" => Ok(NetworkVersion::V9),
+            "v10" => Ok(NetworkVersion::V10),
+            "v11" => Ok(NetworkVersion::V11),
+            "v12" => Ok(NetworkVersion::V12),
+            "v13" => Ok(NetworkVersion::V13),
+            "v14" => Ok(NetworkVersion::V14),
+            "v15" => Ok(NetworkVersion::V15),
+            _ => Err(de::Error::custom(&format!(
+                "Invalid network version: {}",
+                version
+            ))),
+        }
+    }
+
+    pub fn serialize<S>(n: &NetworkVersion, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let version_string = match n {
+            NetworkVersion::V0 => "V0",
+            NetworkVersion::V1 => "V1",
+            NetworkVersion::V2 => "V2",
+            NetworkVersion::V3 => "V3",
+            NetworkVersion::V4 => "V4",
+            NetworkVersion::V5 => "V5",
+            NetworkVersion::V6 => "V6",
+            NetworkVersion::V7 => "V7",
+            NetworkVersion::V8 => "V8",
+            NetworkVersion::V9 => "V9",
+            NetworkVersion::V10 => "V10",
+            NetworkVersion::V11 => "V11",
+            NetworkVersion::V12 => "V12",
+            NetworkVersion::V13 => "V13",
+            NetworkVersion::V14 => "V14",
+            NetworkVersion::V15 => "V15",
```

Comment:
Deserializer uses `"v15"` and serializer a capital `"V15"` - is that on purpose?

Reply:
the deserializer converts it to lowercase before matching against network versions. Just makes it so v10 and V10 are valid